src/Microsoft.VisualStudio.Editors/PropPages/ReferencePropPage.vb

```diff
@@ -792,10 +793,11 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 Dim solution = visualStudioWorkspace.CurrentSolution
                 Dim CurrentImportNames As New List(Of String)
 
-                For Each projectId In solution.ProjectIds
-                    ' We need to find the project that matches by IVsHierarchy
-                    If visualStudioWorkspace.GetHierarchy(projectId) Is ProjectHierarchy Then
-                        Dim compilationTask = solution.GetProject(projectId).GetCompilationAsync(cancellationTokenSource.Token)
+                For Each project In solution.Projects
```

Comment:
> Can you quickly summarize how Roslyn and Project file interplay to populate the imports list?

The ordered way of working of the import list: 

1. Populate the list from CompilationOptions, that comes from VisualStudioWorkspace ->Project -> Compilation
2. Get the entire list of Namespaces that the project can access from VisualStudioWorkspace -> Project -> Compilation
3. Do the calculation to display the list in the UI
4. User Action, say Remove an existing import.
5. Action is validated by **_<OurComponent>_**. _OurComponent_ is VisualStudioVSImports here.
6. On Validation success, remove the import through MSBuild Evaluation. At this point, the control goes back to the References Property Page.
7. References Property Page recognizes that some import change has been done and does step 1,2 and 3 again to refresh the list.

This is where the problem arises. 

For the refresh to be successful, the change performed on the MSbuild Evaluation project should reach VisualStudioWorkspace. This is because, Populating of the list(initialization or refresh) depend on VisualStudioWorkspace but not the previous state of the populated list. Since Global imports are part of options, this change is not pushed to Roslyn until a Design time build happens and since we dont have a design time build happening between the step 6 and 7, the refresh operation populates the UI with stale information.